tfhe/src/core_crypto/algorithms/lwe_keyswitch.rs

```diff
@@ -315,6 +316,121 @@ pub fn keyswitch_lwe_ciphertext_other_mod<Scalar, KSKCont, InputCont, OutputCont
     }
 }
 
+/// Keyswitch an [`LWE ciphertext`](`LweCiphertext`) with a certain InputScalar type to represent
+/// data encrypted under an [`LWE secret key`](`LweSecretKey`) to another [`LWE secret
+/// key`](`LweSecretKey`) using a different OutputScalar type to represent data.
+///
+/// # Notes
+///
+/// This function only supports power of 2 moduli and going from a large InputScalar with
+/// `input_bits` to a a smaller OutputScalar with `output_bits` and `output_bits` < `input_bits`.
```

Comment:
Why this constraint?
In my understanding, we only need `level*base_log<=input_bits` (and arguably `base_log < output_bits`)

Reply:
It was to make things easier when managing casting from input to output

Basically what we are interested in at the moment is going from 64 bits to 32 bits during the keyswitch

When I was writing the code it looked like there would be more and more possible branches in the implementation if I made the most generic version so as a first version of that I thought of simplifying the function to match our need with the intent to add more functionality later if required.

What do you think ?

You are right on the required level * base_log < output_bits that I did not capture here.

Reply:
but then we likely have the same input scalar and people should use the other implem